File: examples/mock-app.ts
#!/usr/bin/env bun
// mock-app — drives Acceptance Criterion 4 manual verification.
//
// Boots the SDK pointed at $BROKER_HOST:$BROKER_PORT (defaults
// 127.0.0.1:1883), publishes retained presence, holds open the heartbeat
// loop, and gracefully stops on SIGTERM / SIGINT (clears retained
// presence, then exits 0).
//
// Run from sdk-node/:
//   bun run examples/mock-app.ts
//   BROKER_HOST=192.168.1.10 BROKER_PORT=1883 bun run examples/mock-app.ts

import { Inspire } from '../src/index'

async function main(): Promise<void> {
  const host = process.env.BROKER_HOST ?? '127.0.0.1'
  const port = Number(process.env.BROKER_PORT ?? 1883)
  const slug = process.env.MOCK_APP_SLUG ?? 'mock-app'
  const version = process.env.MOCK_APP_VERSION ?? '0.1.0'

  // eslint-disable-next-line no-console
  console.log(`[mock-app] connecting to ${host}:${port} as ${slug} v${version}`)

  const client = await Inspire.start({
    slug,
    version,
    broker: { host, port },
  })

  await client.setStatus('ready', 'mock app online')
  // eslint-disable-next-line no-console
  console.log(`[mock-app] online as ${client.slug}|${client.nodeId} — Ctrl-C to stop`)

  const shutdown = async (sig: string) => {
    // eslint-disable-next-line no-console
    console.log(`[mock-app] received ${sig}, clearing retained presence and exiting`)
    try {
      await client.setStatus('stopping', `received ${sig}`)
    } catch {
      /* broker may already be unreachable */
    }
    await client.stop()
    process.exit(0)
  }
  process.on('SIGTERM', () => void shutdown('SIGTERM'))
  process.on('SIGINT', () => void shutdown('SIGINT'))

  // Keep process alive — the heartbeat interval already does this, but
  // make it explicit so a future SDK refactor that uses setImmediate
  // doesn't break the example.
  setInterval(() => {
    /* idle */
  }, 60_000)
}

void main().catch((err) => {
  // eslint-disable-next-line no-console
  console.error('[mock-app] fatal:', err)
  process.exit(1)
})

File: sdk-python/examples/mock_app.py
"""Mock inspire-* app for manual end-to-end testing of atrium's Running view.

Run with:

    python examples/mock_app.py

Or with a non-default broker host:

    INSPIRE_BROKER_HOST=192.168.1.10 python examples/mock_app.py

Mirrors sdk-node/examples/mock-app.ts. Atrium's Running view should display
this row within 2s of launch (spec AC 4) and evict it within 60s of SIGKILL
(AC 5) or within ~2s of graceful Ctrl-C (LWT-cleared).
"""

from __future__ import annotations

import os
import signal
import sys
import threading
import time

from inspire_sdk import Inspire


def main() -> int:
    host = os.environ.get("INSPIRE_BROKER_HOST", "127.0.0.1")
    port = int(os.environ.get("INSPIRE_BROKER_PORT", "1883"))
    print(f"connecting to {host}:{port} ...", flush=True)

    client = Inspire.start(
        slug="mock-app-py",
        version="0.1.0",
        broker={"host": host, "port": port},
    )
    print(f"connected; node_id={client.node_id}", flush=True)
    client.set_status("ready", "mock app py running")

    stop = threading.Event()

    def handle_sigint(_signum, _frame):
        print("received SIGINT, clearing retained presence and exiting", flush=True)
        stop.set()

    def handle_sigterm(_signum, _frame):
        print("received SIGTERM, clearing retained presence and exiting", flush=True)
        stop.set()

    signal.signal(signal.SIGINT, handle_sigint)
    signal.signal(signal.SIGTERM, handle_sigterm)

    client.on_command(
        "shutdown",
        lambda _msg: stop.set(),
    )

    while not stop.wait(1.0):
        pass

    client.stop()
    return 0


if __name__ == "__main__":
    sys.exit(main())
